meminfo.go

```diff
@@ -162,114 +217,176 @@ func parseMemInfo(r io.Reader) (*Meminfo, error) {
 	var m Meminfo
 	s := bufio.NewScanner(r)
 	for s.Scan() {
-		// Each line has at least a name and value; we ignore the unit.
 		fields := strings.Fields(s.Text())
-		if len(fields) < 2 {
-			return nil, fmt.Errorf("%w: Malformed line %q", ErrFileParse, s.Text())
-		}
+		var val, valBytes uint64
 
 		v, err := strconv.ParseUint(fields[1], 0, 64)
 		if err != nil {
 			return nil, err
 		}
 
+		switch len(fields) {
+		case 2:
+			// No unit present, use the parsed the value as bytes directly.
+			val = v
+			valBytes = v
+		case 3:
+			// Unit present in optional 3rd field, convert and normalize to bytes.
+			bytes, err := humanize.ParseBytes(fmt.Sprintf("%s %s", fields[1], fields[2]))
```

Comment:
`humanize.ParseBytes()` is producing incorrect results as it thinks `kB` is 1000. The kernel values for this are KiB (1024).

The kernel is [hard-coded to only have one representation](https://github.com/torvalds/linux/blob/3aba70aed91f2b283f7952be152ad76ec5c34975/fs/proc/meminfo.c#L28-L32), which is `kB`.

Let's leave out the `humanize` package and simply check for the string `kB` in the fields. If the extra field is different we can return a parse error.

Reply:
Ah, I didn't realize the kernel only had a single unit available for the optional field -- I didn't see direct docs on kernel.org related to the unit field for meminfo, and the docs I did find read to me as if other units were available. Given the misinterpretation of the unit itself and how static the underlying data is, it seems fair to just handle the `kB` case. 

Rebased and then noticend a new release just came out, so refactored on top of current master as well. I think it's ready for another look :+1: